plugins/chat/assets/javascripts/discourse/services/chat-channels-manager.js

```diff
@@ -0,0 +1,137 @@
+import Service, { inject as service } from "@ember/service";
+import Promise from "rsvp";
+import ChatChannel from "discourse/plugins/chat/discourse/models/chat-channel";
+import { tracked } from "@glimmer/tracking";
+
+const DIRECT_MESSAGE_CHANNELS_LIMIT = 20;
+
+export default class ChatChannelsManager extends Service {
+  @service chatSubscriptionsManager;
+  @service chatApi;
+  @service currentUser;
+  @tracked _cached = {};
```

Comment:
I notice you're doing a lot of `this.notifyPropertyChange("_cache")`. Instead, you might like to use a `TrackedObject`, which automatically tracks all changes to its keys/values.

https://github.com/tracked-tools/tracked-built-ins

https://github.com/discourse/discourse/blob/2e224530575fccc218e96fa62f718e479dfad7b3/app/assets/javascripts/discourse/app/services/site-settings.js#L8-L8

Reply:
oh nice, didn't know this new thing, awesome!